src/demo/actions/counter.ts

```diff
@@ -8,17 +8,26 @@ Code distributed by Google as part of the polymer project is also
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 */
 
+import * as r from 'redux';
 export const INCREMENT = 'INCREMENT';
 export const DECREMENT = 'DECREMENT';
 
+export interface CounterActionIncrement extends r.Action<'INCREMENT'> {};
+export interface CounterActionDecrement extends r.Action<'DECREMENT'> {};
+export type CounterAction = CounterActionIncrement | CounterActionDecrement;
+
 export const increment = () => {
-  return {
+  const action: CounterActionIncrement = {
```

Comment:
Style note, you can get the same result without introducing a local by annotating the function with the return type. i.e. `export const increment = (): CounterActionIncrement => {`